### PR TITLE
ECO-2788: Refund amount is taken from item instead of refund module

### PR DESCRIPTION
**Pull request by lockw1n**
- Developer(s): @aleksey-kotsuba

- Ticket: https://spryker.atlassian.net/browse/ECO-2788

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   CrefoPay              | major (currently 0.0.x)  |                       |

#### Release Notes

* Refund amount is taken from refund transfer.

-----------------------------------------

### Bugfixes
* Refund amount is taken from refund transfer.

### CODE REVIEW

File: src/SprykerEco/Zed/CrefoPay/Business/CrefoPayFacade.php [hunk 1]
```diff
@@ -130,32 +131,36 @@ public function executeCancelOmsCommand(OrderTransfer $orderTransfer, array $sal
      * @api
      *
      * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
-     * @param int $idSalesOrderItem
+     * @param int[] $salesOrderItemIds
      *
      * @return void
      */
-    public function executeCaptureOmsCommand(OrderTransfer $orderTransfer, int $idSalesOrderItem): void
+    public function executeCaptureOmsCommand(OrderTransfer $orderTransfer, array $salesOrderItemIds): void
```
Comment (gechetspr) [inline src/SprykerEco/Zed/CrefoPay/Business/CrefoPayFacade.php:138]: If we have it released already - this is a BC brak

File: src/SprykerEco/Zed/CrefoPay/Business/CrefoPayFacade.php [hunk 2]
```diff
@@ -130,32 +131,36 @@ public function executeCancelOmsCommand(OrderTransfer $orderTransfer, array $sal
...
+     * @param \Generated\Shared\Transfer\RefundTransfer $refundTransfer
      * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
-     * @param int $idSalesOrderItem
+     * @param int[] $salesOrderItemIds
      *
      * @return void
      */
-    public function executeRefundOmsCommand(OrderTransfer $orderTransfer, int $idSalesOrderItem): void
-    {
+    public function executeRefundOmsCommand(
```
Comment (gechetspr) [inline src/SprykerEco/Zed/CrefoPay/Business/CrefoPayFacade.php:156]: same here

File: src/SprykerEco/Zed/CrefoPay/Business/Oms/Command/Builder/RefundOmsCommandRequestBuilder.php [hunk 3]
```diff
@@ -61,18 +60,18 @@ protected function createRefundRequestTransfer(CrefoPayOmsCommandTransfer $crefo
             ->setStoreID($this->config->getStoreId())
             ->setOrderID($crefoPayOmsCommandTransfer->getPaymentCrefoPay()->getCrefoPayOrderId())
             ->setCaptureID($paymentCrefoPayOrderItemTransfer->getCaptureId())
-            ->setAmount($this->createAmountTransfer($paymentCrefoPayOrderItemTransfer))
-            ->setRefundDescription($this->config->getRefundDescription());
+            ->setAmount($this->createAmountTransfer($crefoPayOmsCommandTransfer))
+            ->setRefundDescription($crefoPayOmsCommandTransfer->getRefund()->getComment() ?? $this->config->getRefundDescription());
```
Comment (gechetspr) [inline src/SprykerEco/Zed/CrefoPay/Business/Oms/Command/Builder/RefundOmsCommandRequestBuilder.php]: I would have it as a separate method

**Review by gechetspr — CHANGES_REQUESTED** [review]
Please check my comments.
Also change log should describe everything that you changed in the code. We should not silently change the code without informing users about those changes.

File: src/SprykerEco/Zed/CrefoPay/Business/CrefoPayFacade.php (hunk 1 above)
Comment (lockw1n) [reply src/SprykerEco/Zed/CrefoPay/Business/CrefoPayFacade.php:138]: We haven't released module yet.

File: src/SprykerEco/Zed/CrefoPay/Business/CrefoPayFacade.php (hunk 2 above)
Comment (lockw1n) [reply src/SprykerEco/Zed/CrefoPay/Business/CrefoPayFacade.php:156]: We haven't released module yet.

Comment (gechetspr) [reply src/SprykerEco/Zed/CrefoPay/Business/CrefoPayFacade.php:156]: ok

**Review by gechetspr — APPROVED** [review]

**Review by alex-galych — APPROVED** [review]
The logic for multiple items in refund and capturing will be implemented later

**Pull request merged**